[EDU-6400] devex release notes (#1622)
aziontech/docs: src/content/docs/en/pages/main-menu/release-notes/release-notes.mdx

---
import Tag from 'primevue/tag'

### May 7, 2025

###  Developer Experience Launch

- **Azion Bundler v5.1**: Twice as fast, with optimized build times for major frameworks.  
- **Custom Domain (azion.app)**: Instantly share applications with a readable domain (FQDN) available via the API. Coming soon to Console.  
- **Expanded Integrations**: New CMS platforms (Sanity, Cosmic, ButterCMS).  

These enhancements, along with recently introduced updates to our platform, deliver unprecedented automation, speed, and an elevated developer experience across all Azion Dev Tools.

## May 6, 2025

### Build & Scale: Platform & Product Enhancements

#### Edge Functions

- **WinterTC (ECMA Open Standard)**: Over 80% compatibility with WinterTC APIs, promoting portability and eliminating vendor lock-in.  
- **Runtime Limits Increased**: Edge Functions now run up to 5 minutes of CPU time and support application bundles up to 20 MB.  

> As part of the ongoing Launch Week, we also want to highlight some of the transformative updates recently introduced to our platform: Node.js API Compatibility with [broader support](https://azion-node-matrix-compat.azion.app/) for Node.js APIs, and expanded framework support compatibility with Next.js, Vue, React, Angular, Gatsby, Astro, and more.

Reviewer: ```suggestion
As part of the ongoing Launch Week, we also want to highlight some of the transformative updates recently introduced to our platform: Node.js API compatibility with [broader support](https://azion-node-matrix-compat.azion.app/) for Node.js APIs, and expanded framework support compatibility with Next.js, Vue, React, Angular, Gatsby, Astro, and more.
```

---
import Tag from 'primevue/tag'

### May 7, 2025

###  Developer Experience Launch

- **CLI (GA, cross-platform)**: Faster, intuitive, and now natively available for Windows, macOS, and Linux. Supports resource management, CI/CD integration, and full IaC workflows.  
- **Azion Bundler v5.1**: Twice as fast, optimized build times for major frameworks.  

> - **Azion Lib**: Simplify service interactions with our new libraries. Visit the [reference documentation](https://www.azion.com/en/documentation/products/azion-lib/overview/) for more details.  

Reviewer 2: ```suggestion
- **Azion Lib**: Simplify service interactions with our new libraries. Visit the [reference documentation](/en/documentation/products/azion-lib/overview/) for more details.  
```

- **Git Deployments**: Easy deploys from Git repositories via new pre-configured templates and automatic CI/CD.  
- **Custom Domain (azion.app)**: Instantly share apps with a readable subdomain—Coming soon to Console.  
- **Azion CLI & IaC**: Automate infrastructure with the latest version of command-line provisioning and management, integrated with Infrastructure as Code.
- **Expanded Integrations**: New CMS (Sanity, Cosmic, ButterCMS) platforms.  
- **Deploy to Azion Badge**: One-click deployment from your repo README or docs.  
- **API v4 (GA)**:  
  - New event-driven architecture with decoupled design.  
  - Up to 80% faster critical operations.  
  - JWT/Bearer authentication.  
  - Activity History available via GraphQL.  

>   - "Workload" resource model replaces "Domain"  

Reviewer 2: ```suggestion
  - "Workload" resource model replaces "Domain" .
```

- **Console Improvements**:  
  - Headless, open-source UI.  

>   - Dark mode & new navigation drawers. 

Reviewer 2: ```suggestion
  - Dark mode and new navigation drawers. 
```

  - Real-Time Events: improved filtering and GraphQL queries.

These enhancements deliver unprecedented automation, speed, and developer experience across all Azion Dev Tools.

## May 6, 2025

### Build & Scale: Platform & Product Enhancements

#### Edge Functions

> - **Node.js API Compatibility**: [Broader support](https://azion-node-matrix-compat.azion.app/) for Node.js APIs through new polyfills; including FS, Crypto, Process, HTTP/HTTPS, Events, and Stream modules.  

Reviewer 2: ```suggestion
- **Node.js API Compatibility**: [Broader support](https://azion-node-matrix-compat.azion.app/) for Node.js APIs through new polyfills, including FS, Crypto, Process, HTTP/HTTPS, Events, and Stream modules.  
```

---
import Tag from 'primevue/tag'

### May 7, 2025

> ###  Developer Experience Launch

Reviewer: Aqui vai manter em EN?
```suggestion
###  Developer Experience Launch
```

- **Azion Bundler v5.1:** Twice as fast, optimized build times for major frameworks.  
- **Azion Lib:** Simplify service interactions with our new libraries. Visit the reference [documentation](https://www.azion.com/en/documentation/products/azion-lib/overview/) for more details.  
- **Git Deployments:** Easy deploys from Git repositories via new pre-configured templates and automatic CI/CD.  
- **Custom Domain (azion.app):** Instantly share apps with a readable subdomain—Coming soon to Console.  
- **Azion CLI & IaC:** Automate infrastructure with the latest version of command-line provisioning and management, integrated with Infrastructure as Code.
- **Expanded Integrations:** New CMS (Sanity, Cosmic, ButterCMS) platforms.  
- **Deploy to Azion Badge:** One-click deployment from your repo README or docs.  
- **API v4 (GA):**  
  - New event-driven architecture with decoupled design.  
  - Up to 80% faster critical operations.  
  - JWT/Bearer authentication.  
  - Activity History available via GraphQL.  
  - "Workload" resource model replaces "Domain"  
- **Console Improvements**:  

>   - Headless, open-source UI  
>   - Dark mode & new navigation drawers  
>   - Real-Time Events: improved filtering and GraphQL queries

Reviewer: ```suggestion
  - Headless, open-source UI.  
  - Dark mode & new navigation drawers. 
  - Real-Time Events: improved filtering and GraphQL queries.
```

These enhancements deliver unprecedented automation, speed, and developer experience across all Azion Dev Tools.

## May 6, 2025

### Build & Scale: Platform & Product Enhancements

#### Edge Functions

> - **Node.js API Compatibility:** [Broader suppor](https://azion-node-matrix-compat.azion.app/)t for Node.js APIs through new polyfills; including FS, Crypto, Process, HTTP/HTTPS, Events, and Stream modules.  

Reviewer: ```suggestion
- **Node.js API Compatibility:** [Broader support](https://azion-node-matrix-compat.azion.app/) for Node.js APIs through new polyfills; including FS, Crypto, Process, HTTP/HTTPS, Events, and Stream modules.  
```

- **WinterTC (ECMA Open Standard):** Over 80% compatibility with WinterTC APIs, promoting portability and eliminating vendor lock-in.  
- **Expanded Framework Support:** Now compatible with Next.js, Vue, React, Angular, Gatsby, Astro, and more.  

> - **Runtime Limits Increased:** Edge Functions now run up to 5 minutes of CPU time and support application bundles up to 20MB.  

Reviewer: ```suggestion
- **Runtime Limits Increased:** Edge Functions now run up to 5 minutes of CPU time and support application bundles up to 20 MB.  
```

> - **Templates:** New templates for fast deployment of e-commerce, web sites, blogs, and more—available in Console.

Reviewer: ```suggestion
- **Templates:** New templates for fast deployment of e-commerce, websites, blogs, and more—available in the Console.
```

#### Edge AI

- **AI Model Execution**: Supports LLMs, VLMs, multimodal, embedding, and reranking models on **Edge Runtime**.
- **Model Customization**: Fine-tune models with **Low-Rank Adaptation (LoRA)** for specific business needs.

#### Edge SQL

- **LangChain Vector Store integration**: enables efficient storage and retrieval of vectorial documents.
- **LangChain Vector Retriever integration**: enables advanced hybrid search capabilities for documents, combining the strengths of different search methods for more accurate results.

As part of the ongoing Launch Week, we also want to highlight some of the transformative updates recently introduced to our platform: Vector Search for distributed vector databases and efficient RAG pipelines.

#### Edge Functions

> - **LangGraph Agent Support**: build AI agents with Tool Calling and Agentic RAG, leveraging LangGraph from [LangChain](https://www.langchain.com/).
> - **LangGraph AI Agent Boilerplate**: template to create an agent that can query and interact with data stored in EdgeSQL databases.
> - **Compatibility**: integrates with OpenAI-compatible APIs and LangChain/LangGraph for workflow automation.

Reviewer: ```suggestion
- **LangGraph Agent Support**: Build AI agents with Tool Calling and Agentic RAG, leveraging LangGraph from [LangChain](https://www.langchain.com/).
- **LangGraph AI Agent Boilerplate**: Template to create an agent that can query and interact with data stored in EdgeSQL databases.
- **Compatibility**: Integrates with OpenAI-compatible APIs and LangChain/LangGraph for workflow automation.
```
